### DISCUSSION TITLE
Jax equivalent of tf.scatter_nd

### DISCUSSION THREAD

**Discussion by jpuigcerver** [jax-ml/jax#3658] [General] [answered]
Hi,

Is there a function or short code snippet equivalent to Tensorflow's [scatter_nd](https://www.tensorflow.org/api_docs/python/tf/scatter_nd). 

I know of [jax.lax.scatter](https://jax.readthedocs.io/en/latest/_autosummary/jax.lax.scatter.html), but it has an extra `dimension_numbers` argument which I frankly don't quite understand.

Thanks!

**Comment by shoyer** [comment, marked as answer]
`lax.scatter` exists, but is indeed rather complex. For cases where you would use `tf.scatter_nd`, we recommend using [indexed update functions](https://jax.readthedocs.io/en/latest/jax.ops.html#indexed-update-operators) or the equivalent syntactic sugar using the `.at` property.

The API for indexed updates is very similar but with a different axis order matching NumPy's advanced indexing. To reproduce `scatter_nd` in JAX you could use:
```python
import jax.numpy as jnp

def scatter_nd(indices, updates, shape):
    zeros = jnp.zeros(shape, updates.dtype)
    key = tuple(jnp.moveaxis(indices, -1, 0))
    return zeros.at[key].add(updates)
```

**Reply by jpuigcerver** [reply]
Thank you so much!
